scrunch/mutable_dataset.py

```diff
@@ -0,0 +1,892 @@
+                kwargs[key] = kwargs[key].isoformat()
+
+        return self.resource.edit(**kwargs)
+
+    def delete(self):
+        """
+        Delete a dataset.
+        """
+        self.resource.delete()
+
+    def stream_rows(self, columns):
+        """
+        Receives a dict with columns of values to add and streams them
+        into the dataset. Client must call .push_rows(n) later or wait until
+        Crunch automatically processes the batch.
+
+        Returns the total of rows streamed
+        """
+        importer = Importer()
+        count = len(list(columns.values())[0])
+        for x in range(count):
+            importer.stream_rows(self.resource,
+                                 {a: columns[a][x] for a in columns})
+        return count
+
+    def push_rows(self, count=None):
+        """
+        Batches in the rows that have been recently streamed. This forces
+        the rows to appear in the dataset instead of waiting for crunch
+        automatic batcher process.
+        """
+        if bool(self.resource.stream.body.pending_messages):
+            self.resource.batches.create({
+                'element': 'shoji:entity',
+                'body': {
+                    'stream': count,
+                    'type': 'ldjson'
+                }
+            })
+
+    def fork(self, description=None, name=None, is_published=False,
+             preserve_owner=False, **kwargs):
+        """
+        Create a fork of ds and add virgin savepoint.
+
+        :param description: str, default=None
+            If given, the description to be applied to the fork. If not
+            given the description will be copied from ds.
+        :param name: str, default=None
+            If given, the name to be applied to the fork. If not given a
+            default name will be created which numbers the fork based on
+            how many other forks there are on ds.
+        :param is_published: bool, default=False
+            If True, the fork will be visible to viewers of ds. If False it
+            will only be viewable to editors of ds.
+        :param preserve_owner: bool, default=False
+            If True, the owner of the fork will be the same as the parent
+            dataset. If the owner of the parent dataset is a Crunch project,
+            then it will be preserved regardless of this parameter.
+
+        :returns _fork: scrunch.datasets.Dataset
+            The forked dataset.
+        """
+
+        nforks = len(self.resource.forks.index)
+        if name is None:
+            name = "FORK #{} of {}".format(nforks + 1, self.resource.body.name)
+        if description is None:
+            description = self.resource.body.description
+
+        body = dict(
+            name=name,
+            description=description,
+            is_published=is_published,
+            **kwargs
+        )
+        # not returning a dataset
+        _fork = self.resource.forks.create({"body": body}).refresh()
```

Comment:
Use the shoji_entity helper here, the API will start becoming strict about expecting an `element` key.